### PR TITLE
fix: interceptor fixes to make new call if access_token is not present and not throw access_token errors to sentry

### PR DESCRIPTION
**Pull request by Aniruddha-Shriwant**
## Clickup
https://app.clickup.com/t/86cwqkdza

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling for token refresh processes, improving the stability of user sessions.
	- Streamlined logic for accessing tokens, ensuring a valid token is requested when necessary.

- **Bug Fixes**
	- Resolved issues with error propagation during token refresh, leading to more reliable logout and cache clearing operations.

- **Tests**
	- Updated test cases to verify the correct execution of services during error scenarios without relying on error objects.

### CODE REVIEW

File: src/app/core/interceptors/httpInterceptor.spec.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ import { BehaviorSubject, of, throwError } from 'rxjs';
 import { extendedDeviceInfoMockData, extendedDeviceInfoMockDataWoApp } from '../mock-data/extended-device-info.data';
 import { HttpErrorResponse, HttpHeaders, HttpRequest } from '@angular/common/http';
 
-describe('HttpConfigInterceptor', () => {
+fdescribe('HttpConfigInterceptor', () => {
```
Comment (Chethan-Fyle) [inline src/app/core/interceptors/httpInterceptor.spec.ts:17]: Is `fdescribe()` intentional?

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/core/interceptors/httpInterceptor.spec.ts (hunk 1 above)
Comment (Aniruddha-Shriwant) [reply src/app/core/interceptors/httpInterceptor.spec.ts]: hey yes, I was about to fix this 🤣 
Your speed ⚡

Comment (Aniruddha-Shriwant) [reply src/app/core/interceptors/httpInterceptor.spec.ts]: this is fixed

File: src/app/core/interceptors/httpInterceptor.ts [hunk 2]
```diff
@@ -168,7 +168,7 @@ export class HttpConfigInterceptor implements HttpInterceptor {
                 this.secureStorageService.clearAll();
                 this.storageService.clearAll();
                 globalCacheBusterNotifier.next();
-                return throwError(error);
+                return EMPTY;
```
[the remark this answers is not in the document]
Comment (Aniruddha-Shriwant) [reply src/app/core/interceptors/httpInterceptor.ts:171]: nopes, access_token 401 is not something we should track

File: src/app/core/interceptors/httpInterceptor.ts [hunk 3]
```diff
@@ -67,12 +67,12 @@
...
       concatMap((refreshToken) => this.routerAuthService.fetchAccessToken(refreshToken)),
-      catchError((error) => {
+      catchError(() => {
         this.userEventService.logout();
         this.secureStorageService.clearAll();
         this.storageService.clearAll();
         globalCacheBusterNotifier.next();
-        return throwError(error);
+        return EMPTY;
       }),
```
[the remark this answers is not in the document]
Comment (Aniruddha-Shriwant) [reply src/app/core/interceptors/httpInterceptor.ts:76]: not needed

**Pull request merged**